freqtrade/freqtradebot.py

```diff
@@ -881,7 +881,7 @@ def check_handle_timedout(self) -> None:
 
             fully_cancelled = self.update_trade_state(trade, order)
 
-            if (order['side'] == 'buy' and (
+            if (order['side'] == 'buy' and (order['status'] == 'open' or fully_cancelled) and (
```

Comment:
Note to myself, to remember:
that's (the whole if) is the only thing I need to verify closer (it's getting more and more complicated)
(and for the sell side below correspondingly)

The rest LGTM 👍

Reply:
well don't tell me ...

Letme summarize what it does:

* check side (the rest is identical for both sides, but each with "side" different).
* short-circuit to false if the order is not open and NOT fully cancelled (this is the new part)

inside the or:
* check if it's fully cancelled
* verify timeout
* verify manual timeout call (that's VERY usefull to test timeout stuff, as you can just return "True" all the time) :laughing:

Reply:
```
order['status'] == 'open' or fully_cancelled
```
here `fully_cancelled` is assigned as
```
fully_cancelled = self.update_trade_state(trade, order)
```
and `update_trade_state()` returns True if and only if
```
self.exchange.check_order_canceled_empty(order)
```
returns True, and `check_order_canceled_empty()` is calculated as
```
return order.get('status') in ('closed', 'canceled') and order.get('filled') == 0.0
```

So this part of the condition can be (I do not say it should be) reduced to:
```
(order['status'] == 'open') or (order['status'] in ('closed', 'canceled') and order.get('filled') == 0.0)
```

Just cannot understand for the while why these cases are handled here and the rest (what's the rest of cases? only open and partially filled? or all closed except closed/cancelled and unfilled?)

Reply:
that's the problem - in theory all order-states can show up here (depending on the order type) (open, partially filled, even closed (that's what the change protects from)).

And the last "or" section must only be evaluated in certain conditions.

we had this as 2 nested IF's for clarity a while ago - but it doesn't really make sense as one IF can do the same thing.

Yes - the "filled_empty" could be done here too (instead of using the return value of `update_trade_state` - but since it's needed in `update_trade_state` too, that would be rather ... pointless.

I've renamed the variable to make that part clearer ...

Reply:
LGTM if it fixes this spamming issue 👍 
I'll review the logic here in more details later.

Some comments could be added here, explaining these hard for perception conditions...